### ISSUE TITLE
Possible to create invalid Scan with negative oscillation

### ISSUE THREAD

**Issue by dagewa** [cctbx/cctbx_project#206]
This comes from a user, who has SMV format images with `OSC_RANGE=-0.10;` in the header, which ends up as a scan with a negative oscillation (rather than a positive oscillation and an inverted rotation axis, which is the usual way of handling this).
```
Scan:
    image range:   {0,1060}
    oscillation:   {63.39,-0.1}
    exposure time: 0.096288
```
It seems that scans with a negative oscillation are invalid. For example, make a scan with negative oscillation:
```
from dxtbx.model import ScanFactory
scan=ScanFactory.make_scan(image_range = (1,90), oscillation=(0, -1.0),
  exposure_times=0.1, epochs=range(90), deg=True)
```
Now, use `scan.is_angle_valid` to check the range of valid angles between -360 and +360 degrees. What you find is that [-360,-90] ∪ [0, 270] is reported as _valid_. So it looks like a 270 degree scan rather than a 90 degree scan.

By contrast, set up a normal scan with positive oscillation:
```
scan=ScanFactory.make_scan(image_range = (1,90), oscillation=(0, 1.0),
  exposure_times=0.1, epochs=range(90), deg=True)
```
Now the `is_angle_valid` check tells us that [-360, -270] ∪ [0, 90] is valid, as expected.

**Comment by dagewa** [comment]
Rather than fixing `is_angle_valid` to allow negative oscillation, my personal preference would be to disallow negative oscillation at all in `Scan`, so that situation is always handled by inverting the rotation axis. Allowing negative oscillation adds complication, and it may cause problems in other parts of the code. At least in this case the problem would have been caught at import, rather than causing a (silent) failure in refinement during indexing.

**Comment by jmp1985** [comment]
I agree. Lets fix the format class to invert the rotation axis and add a check in the scan constructor to throw an exception if the angle is negative.

**Comment by dagewa** [comment]
Reopened for discussion. Should `Scan` allow negative oscillations?

Reasons against:
- The `Scan` object currently has inconsistent behaviour with negative oscillations
- DIALS would require changes to allow processing with negative oscillations (unclear if trivial fixes only or not)
- We have a preferred mechanism for handling such cases at import, which is to invert the rotation axis

Reasons for:
- Some image headers really do have negative oscillation, so allowing it produces a better match between `dxtbx.print_header` output and the header data
- Adding the logic to invert rotation axis and take the absolute value of the oscillation angle for all possibly affected formats is some work (particularly for multi axis goniometers)
- If automatically inverting the rotation axis, then the `Goniometer` model depends on the scan direction. Thus comparing equality of goniometers between two rotation experiments that simply turn in opposite directions will produce `False`, which may be unexpected.
